src/System.Net.Security/src/System/Net/SslStreamPal.Windows.cs

```diff
@@ -126,8 +127,13 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certi
             return AcquireCredentialsHandle(direction, secureCredential);
         }
 
-        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] writeBuffer, int size, int headerSize, int trailerSize, out int resultSize)
+        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] input, int offset, int size, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
         {
+            byte[] writeBuffer = output;
+
+            // Copy the input into the output buffer to prepare for SCHANNEL's expectations
+            Buffer.BlockCopy(input, offset, writeBuffer, headerSize, size);
+
```

Comment:
Schannel controls the ciphersuite selection during the TLS handshake. The .Net APIs only allow the selection of TLS protocol versions.
It's not feasible to reproduce this in a functional test but it could be reproducible in a unit-test. The work isn't simple as it would require a new mock-implementation of either the PAL or even the Schannel/OpenSSL APIs.